src/StringUtils.cpp

```diff
@@ -14,6 +14,74 @@
 #include <numeric>
 #include <sstream>
 
+
+template<>
+std::string NAS2D::from_string(const std::string& value)
+{
+	return value;
+}
+
+template<>
+bool NAS2D::from_string(const std::string& value)
+{
+	auto v = toLowercase(value);
+	return v == "true";
+}
+
+template<>
+int NAS2D::from_string(const std::string& value)
+{
+	return std::stoi(value);
+}
+
+template<>
+unsigned int NAS2D::from_string(const std::string& value)
+{
+	return static_cast<unsigned int>(std::stoul(value));
```

Comment:
This may do silent wrapping of values that are out of range.

Reply:
The STL doesn't have an overload for `std::stoui`, unfortunately. This was the only way to safely  coerce the value into an unsigned integer. Using `std::stoi` drops half the values in the range of `unsigned int`. Getting the value as an `unsigned long` at least covers all of `unsigned int` and the value can be downcast.

`Dictionary::set(..., unsigned int)` will force the input to only be within the acceptable range of `unsigned int`. Converting an `unsigned int` to `unsigned long` and back is safe to do.

Reply:
Yes, though we have no guarantee the `set` and `get` calls will be properly paired on type.

It might be wise to check the results with `numeric_limits<unsigned int>::max()`, and `throw` if it is out of range. Once the check is complete, the `static_cast` should be safe to do.